components/fxa-client/src/oauth.rs

```diff
@@ -138,6 +138,29 @@ impl FirefoxAccount {
         self.oauth_flow(url, &scopes)
     }
 
+    /// Fetch an OAuth code for a particular client using a session token from the account state.
+    /// This method doesn't support OAuth public clients at this time.
+    ///
+    /// * `client_id` - OAuth client id.
+    /// * `scopes` - Space-separated list of requested scopes.
+    pub fn authorized_code_using_session_token(
+        &mut self,
+        client_id: &str,
+        scope: &str,
+    ) -> Result<String> {
+        let state = util::random_base64_url_string(16)?;
```

Comment:
@rfk do we want state as a param here or just leave it random for now?

Reply:
I would expect callers to always want to pass in a `state` parameter - they're going to take the resulting code and use it to complete an oauth flow somewhere, and completing an OAuth flow involves checking against a known `state` value. We're not providing any value to the caller by synthesizing a `state` here for them.